docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md

```diff
@@ -0,0 +1,183 @@
+
+- It causes agent container stop on ECS with a given `CLUSTER_NAME` using an SSM docs for a certain chaos duration.
+
+:::tip Fault execution flow chart
+![ECS Agent Stop](./static/images/ecs-agent-stop.png)
+:::
+
+## Uses
+
+<details>
+<summary>View the uses of the experiment</summary>
+<div>
+Agent chaos stop is another very common and frequent scenario we find with ECS clusters that can result in breaking of agent that manages task container on ECS cluster and impact its delivery. Such scenarios that can still occur despite whatever availability aids docker provides.
+
+Killing the agent container will distrupt the performance of it and impact to smooth working of task containers. So this category of chaos experiment helps to build the immunity on the application undergoing any such scenarios.
+
+</div>
+</details>
+
+## Prerequisites
+
+:::info
+
+- Ensure that Kubernetes Version >= 1.17
+
+**AWS EC2 Access Requirement:**
+
+- Ensure that the ECS container metadata is enabled this feature is disabled by default. To enable it please follow the aws docs to [Enabling container metadata](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html). If you have your task running prior this activity you may need to restart it to get the metadata directory as mentioned in the docs.
+
+- Ensure that both you and ECS cluster instances have a Role with required AWS access to do SSM and ECS operations. Refer the below mentioned sample policy for the experiment (please note that the sample policy can be minimised further). To know more checkout [Systems Manager Docs](https://docs.aws.amazon.com/systems-manager/latest/userguide/setup-launch-managed-instance.html). Also, please refer the below mentioned policy required for the experiment.
+
+- Ensure to create a Kubernetes secret having the AWS access configuration(key) in the `CHAOS_NAMESPACE`. A sample secret file looks like:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cloud-secret
+type: Opaque
+stringData:
+  cloud_config.yml: |-
+    # Add the cloud AWS credentials respectively
+    [default]
+    aws_access_key_id = XXXXXXXXXXXXXXXXXXX
+    aws_secret_access_key = XXXXXXXXXXXXXXX
+```
+
+- If you change the secret key name (from `cloud_config.yml`) please also update the `AWS_SHARED_CREDENTIALS_FILE` ENV value on `experiment.yaml`with the same name.
+
+## Default Validations
+
+:::info
+
+- ECS container instance should be in healthy state.
+
+:::
+
+## Experiment tunables
+
+<details>
+    <summary>Check the Experiment Tunables</summary>
+    <h2>Mandatory Fields</h2>
+    <table>
+        <tr>
+        <th> Variables </th>
+        <th> Description </th>
+        <th> Notes </th>
+        </tr>
+        <tr> 
+        <td> CLUSTER_NAME </td>
+        <td> Name of the target ECS cluster</td>
+        <td>  </td>
```

Comment:
We can add an example in the notes section (or default value) to maintain consistency in docs.